### PR TITLE
Pre-Commit: Run PHP lint before commit

### PR DESCRIPTION
**Pull request by kraftbj**
I committed a PHP syntax error. While the unit testing caught it, we shouldn't rely on that. This would add a precommit action to lint if there were changed PHP files.

#### Changes proposed in this Pull Request:

* Add PHP lint on precommit.

#### Jetpack product discussion

n/a

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* On this branch, edit a PHP file with some type of bad syntax.
* Try to commit. Ensure it fails to commit.

### CODE REVIEW

File: tools/git-hooks/pre-commit-hook.js [hunk 1]
```diff
@@ -198,6 +198,26 @@ function runEslintChanged( toLintFiles ) {
...
+	const phpcsLintResult = spawnSync( 'composer', [ 'php:lint', ...toLintFiles ], {
+		shell: true,
+		stdio: 'inherit',
+	} );
+
+	if ( phpcsLintResult && phpcsLintResult.status ) {
+		checkFailed( 'PHPCS found linting/syntax errors!\n' );
+		exit( exitCode );
+	}
+}
```
Comment (anomiex) [inline tools/git-hooks/pre-commit-hook.js:219]:
```suggestion
/** Run php:lint
 *
 * @param {Array} toLintFiles - List of files to lint
 */
function runPHPLinter( toLintFiles ) {
	if ( ! toLintFiles.length ) {
		return;
	}

	const phpLintResult = spawnSync( 'composer', [ 'php:lint', ...toLintFiles ], {
		shell: true,
		stdio: 'inherit',
	} );

	if ( phpLintResult && phpLintResult.status ) {
		checkFailed( 'PHP found linting/syntax errors!\n' );
		exit( exitCode );
	}
}
```

File: tools/git-hooks/pre-commit-hook.js [hunk 2]
```diff
@@ -345,6 +365,8 @@ if ( phpFiles.length > 0 ) {
 		shell: true,
 		stdio: 'inherit',
 	} );
+
+	runPHPCSLinter( phpFiles );
```
Comment (anomiex) [inline tools/git-hooks/pre-commit-hook.js:369]:
```suggestion
	runPHPLinter( phpFiles );
```

Also I'm not convinced this is the best place for this. I'd move it up to around line 361, I think.

**Review by anomiex — COMMENTED** [review]
I guess we accidentally stopped doing this with #17729. Good to add it back.

I suggested changes to clean up the naming inline. I also suggest moving the call to the new check up slightly.

**Comment by kraftbj** [comment]
@anomiex I think unrelated to this PR is the `PR is up-to-date` failure. I've ensured this branch is up to date with `master`, so I think the check itself is failing, not that this branch is in an error condition.

**Comment by anomiex** [comment]
Yeah, now that that PR is merged I have to make sure it actually works and fix whatever is wrong. First guess is that we haven't actually created the tag yet, so it fails trying to locate it.

**Review by anomiex — APPROVED** [review]
Looks good to me.

**Pull request merged**